src/_pytest/mark/structures.py

```diff
@@ -276,19 +295,24 @@ def test_function():
     2. If called with a single function as its only positional argument and
        no additional keyword arguments, it attaches the mark to the function,
        containing all the arguments already stored internally in the
-       MarkDecorator.
+       ``MarkDecorator``.
 
-    3. When called in any other case, it returns a new MarkDecorator instance
-       with the original MarkDecorator's content updated with the arguments
-       passed to this call.
+    3. When called in any other case, it returns a new ``MarkDecorator``
+       instance with the original ``MarkDecorator``'s content updated with
+       the arguments passed to this call.
 
-    Note: The rules above prevent MarkDecorators from storing only a single
-    function or class reference as their positional argument with no
+    Note: The rules above prevent a ``MarkDecorator`` from storing only a
+    single function or class reference as its positional argument with no
     additional keyword or positional arguments. You can work around this by
     using `with_args()`.
     """
 
-    mark = attr.ib(type=Mark, validator=attr.validators.instance_of(Mark))
+    mark: Mark
```

Comment:
We lost the runtime validation, but I guess this is intentional given that it is an internal class and mypy should be able to catch any incorrect arguments here, correct?

Reply:
im ok with that bit, before mypy this helped find a bug as far as i recall, these days mypy is the better choice (imho)

Reply:
Right, I don't think we do such checks elsewhere much.. The practical reason I removed it is that it seems that with `init=False` it doesn't work anymore (though I'm not sure), and it didn't seem worth the bother to add it to the `__init__`.